src/aleph/web/controllers/storage.py

```diff
@@ -230,19 +337,27 @@ async def storage_add_file(request: web.Request):
     signature_verifier = get_signature_verifier_from_request(request)
     config = get_config_from_request(request)
     grace_period = config.storage.grace_period.value
-
-    post = await request.post()
-    try:
-        file_field = post["file"]
-    except KeyError:
-        raise web.HTTPUnprocessableEntity(reason="Missing 'file' in multipart form.")
-
-    if isinstance(file_field, FileField):
-        uploaded_file: UploadedFile = MultipartUploadedFile(file_field, len(file_field.file.read()))
+    metadata = None
+    uploaded_file: Optional[UploadedFile] = None
+
+    if request.content_type == "multipart/form-data":
+        reader = await request.multipart()
+        async for part in reader:
+            if part.name == 'file':
+                uploaded_file = MultipartUploadedFile(part, MAX_FILE_SIZE)
+                await uploaded_file.read_and_validate()
+            elif part.name == 'metadata':
+                metadata = await part.read(decode=True)
     else:
-        uploaded_file = RawUploadedFile(file_field)
+        uploaded_file = RawUploadedFile(request=request, max_size=MAX_UNAUTHENTICATED_UPLOAD_FILE_SIZE)
+        await uploaded_file.read_and_validate()
 
-    metadata = post.get("metadata")
+    if uploaded_file is None:
+        raise web.HTTPBadRequest(reason="No file uploaded")
```

Comment:
This one looks like a hard error to debug with very little indication of the effective problem :/